### PR TITLE
Add tls support

### PR DESCRIPTION
**Pull request by brandonbodnar-wk**
_Upstream PR at https://github.com/github/gh-ost/pull/705_

Description
-----------

This PR adds initial support for enforcing TLS/SSL connections for the databases involved in the schema changes. In this initial pass, a gh-ost user can enforce ssl be used for connections using the `--ssl` flag on the command line. If the user wishes to enforce checking validity of the database's certificate, the user can also specify a CA certificate in pem format with the `--ssl-ca` option.

Adding these two options is sufficient for our use case right now in connecting to [AWS RDS instances](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_MySQL.html#MySQL.Concepts.SSLSupport). While there are plenty more options that can be specified by the native mysql client regarding ssl, this can provide a good starting point for other work by those with more complex needs.

Testing
---------
We tested this updated code against an Aurora2 cluster running a MySQL 5.7 compatible engine.

- For testing we added a user to the mysql cluster with REQUIRE SSL set.
- We then attempted to run an alter table command using gh-ost against the cluster, but without the new ssl options enabled. As expected we received an access denied when attempting to connect with `gh-ost`.
- We then attempted the same alter table command using all the same options, except also adding on the new `--ssl` flag, along with specifying the `--ssl-ca` certificate for AWS RDS. The job executed flawlessly.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by ericklaus-wf — APPROVED** [review]
Same commits we previously reviewed.

**Review by ericklaus-wf — APPROVED** [review]
+10 This worked on wk-dev when we turned on require SSL.

**Comment by teresarevious-wf** [comment]
RM +1 dependencies won't be scanned for this repo. this is a temporary fork/fix for tls, once the PR into the original fork is merged that will be used again.

**Comment by matthewbelisle-wf** [comment]
Don't merge this yet @brandonbodnar-wk because once https://github.com/Workiva/gh-ost/pull/3 is merged the builds will pass on this.

File: go/mysql/connection.go
Comment (matthewbelisle-wf) [inline go/mysql/connection.go:80]: @brandonbodnar-wk Can you tweak the logic here so that if it uses ssl, it verifies the cert by default?

Comment (matthewbelisle-wf) [reply go/mysql/connection.go:80]: (Like how curl only skips cert verification if you pass a `--insecure` flag.)

Comment (matthewbelisle-wf) [reply go/mysql/connection.go:80]: cc @ericklaus-wf

Comment (brandonbodnar-wk) [reply go/mysql/connection.go:80]: @matthewbalvanz-wf That is not the default behavior for mysql. This (calling `--ssl` in on the command line) attempts to emulate the behavior of `--ssl-mode=REQUIRED` in [5.7](https://dev.mysql.com/doc/refman/5.7/en/encrypted-connection-options.html#option_general_ssl-mode), which is the same as `--enable-ssl` for [mysql 5.6](https://dev.mysql.com/doc/refman/5.7/en/encrypted-connection-options.html#option_general_ssl). That setting only requires that the connection by encrypted with no guarantees about the certificate provided by the server is from a valid CA (self-signed certificates are okay).

Those (like us) who want additional guarantees that the cert is signed by a specific CA authority, can use the additional `--ssl-ca` flag to behave like `--ssl-mode=VERIFY_CA` from the mysql command line client.

Comment (brandonbodnar-wk) [reply go/mysql/connection.go:80]: The `caCertificatePath` is set using that flag.

Comment (matthewbelisle-wf) [reply go/mysql/connection.go:80]: Hmm I see what you're saying @brandonbodnar-wk , but it seems best to make that `--ssl` flag emulate `VERIFY_CA` instead, unless an `--ssl-insecure` flag is specifically set. The rest of the infosec team agrees that having ssl without verification by default is not a good idea. Here's a PR that achieves what I'm thinking.

https://github.com/Workiva/gh-ost/pull/4

Can you check that out and merge it if it looks good? Once that's merged then I'll +1 this PR and approve it in the RM console.

Comment (brandonbodnar-wk) [reply go/mysql/connection.go:80]: Thanks @matthewbalvanz-wf. I've made a few tweaks to that PR and merged it here.

**Comment by matthewbelisle-wf** [comment]
Security +1

**Review by ericklaus-wf — APPROVED** [review]

**Comment by teresarevious-wf** [comment]
RM +1

**Pull request merged**
